tensorboard/webapp/metrics/views/main_view/card_grid_test.ts

```diff
@@ -0,0 +1,188 @@
+        [groupName]="cardObserver"
+      ></metrics-card-grid>
+      <div class="placeholder">placeholder</div>
+    </div>
+  `,
+  styles: [
+    `
+      div {
+        position: fixed;
+        display: block;
+        height: ${scrollElementHeight}px;
+        overflow-y: scroll;
+      }
+      .placeholder {
+        position: relative;
+        height: 700px;
+        display: block;
+      }
+    `,
+  ],
+})
+class TestableComponent {
+  @Input() groupName: string = 'test group name';
+  @Input() cardIdsWithMetadata: CardIdWithMetadata[] = [
+    {
+      cardId: 'card1',
+      plugin: PluginType.SCALARS,
+      tag: 'tagA',
+      runId: null,
+    },
+    {
+      cardId: 'card2',
+      plugin: PluginType.SCALARS,
+      tag: 'tagA/Images',
+      runId: 'run1',
+      sample: 0,
+    },
+    {
+      cardId: 'card3',
+      plugin: PluginType.SCALARS,
+      tag: 'tagB/meow/cat',
+      runId: 'run1',
+      sample: 0,
+    },
+  ];
+  @Input() cardObserver: CardObserver = new CardObserver();
+}
+
+describe('card grid', () => {
+  let store: MockStore<State>;
+  beforeEach(async () => {
+    await TestBed.configureTestingModule({
+      imports: [NoopAnimationsModule, ScrollingModule],
+      declarations: [
+        CardGridComponent,
+        CardGridContainer,
+        TestableComponent,
+        CardViewComponent,
+        CardViewContainer,
+        CardLazyLoader,
+      ],
+      providers: [
+        provideMockStore({
+          initialState: appStateFromMetricsState(),
+        }),
+      ],
+    }).compileComponents();
+
+    store = TestBed.inject<Store<State>>(Store) as MockStore<State>;
+    store.overrideSelector(selectors.getRunColorMap, {});
+  });
+
+  fit('keeps pagination button position when page size changes', async () => {
+    store.overrideSelector(settingsSelectors.getPageSize, 2);
+    let scrollOffset = Math.floor(Math.random() * scrollElementHeight);
+    const fixture = TestBed.createComponent(TestableComponent);
+    fixture.detectChanges();
+    const nextButtons = fixture.debugElement
+      .queryAll(By.css('.next'))
+      .map((nextDebugElements) => {
+        return nextDebugElements.nativeElement!;
+      });
+    const previousButtons = fixture.debugElement
+      .queryAll(By.css('.prev'))
+      .map((nextDebugElements) => {
+        return nextDebugElements.nativeElement!;
+      });
+    const PaginationInput: HTMLInputElement = fixture.debugElement.query(
+      By.css('input')
+    ).nativeElement;
+    const scrollingElement = fixture.nativeElement.children[0];
+
+    // Test scrolling adjustements on bottom next button.
+    scrollingElement.scrollTo(0, nextButtons[1].offsetTop - scrollOffset);
+    fixture.detectChanges();
+    nextButtons[1].click();
+    fixture.detectChanges();
+    // Clear call stack to invoke the scroll adjustement logic.
+    await timeout(0);
+    expect(nextButtons[1].offsetTop - scrollingElement.scrollTop).toEqual(
+      scrollOffset
+    );
```

Comment:
Great; would you be able to assert and make sure that the content size has changed after clicking on the next button? That is, this test would pass as is if content in the grid never changed.

Reply:
I didn't like the idea of asserting something that we actually do not really care much about. However, I see your point that we are not sure the test is working properly if we do not have it.

I added expect calls for this where appropriate.